OracleGoldenGate/21c/bin/deployment-main.sh

```diff
@@ -125,34 +130,69 @@ function setup_deployment_directories() {
 }
 
 ##
-##  r u n _ p r e s t a r t _ c o m m a n d
-##
-## Hook for launching custom command in the container prior to ogg start
-## If defined, the command identified by ${RUN_COMMAND} will be run.
-## 
-## The value of ${RUN_COMMAND_BACKGROUND} by default is set to true.
-##     When ${RUN_COMMAND_BACKGROUND} is true:
-##          The command identified by ${RUN_COMMAND} will be run in the background 
-##          and its return status is ignored
-##     When ${RUN_COMMAND_BACKGROUND} is false:
-##          The command identified by ${RUN_COMMAND} will be run in the foreground 
-##          and must return a status of 0 in order to proceed with the start sequence
-##
-function run_prestart_command {
-    [[ ! -z "${RUN_COMMAND}" ]] && {
-
-        case "${RUN_COMMAND_BACKGROUND,,}" in
-            true)
-                ${RUN_COMMAND} &
-                ;;
-
-            false)
-                ${RUN_COMMAND} || abort "${RUN_COMMAND} did not complete successfully!"
-                ;;
-            *)
-                abort "Unknown RUN_COMMAND_BACKGROUND state [${RUN_COMMAND_BACKGROUND}] for run_prestart_command: expected true/false"
-        esac
+##  r u n _ u s e r _ s c r i p t s
+##
+## Hook for launching custom scripts in the container before and after ogg start
+##     Default Values:
+##       - ${ABORT_ON_USER_SCRIPT_ERRORS} : true
+##       - ${SETUP_USER_SCRIPTS}          : "${OGG_HOME}/scripts/setup"
+##       - ${STARTUP_USER_SCRIPTS}        : "${OGG_HOME/scripts/startup}"
+##
+## Scripts are run lexicographically and recursively from the directories pointed to by:
+##      - ${SETUP_USER_SCRIPTS} are executed prior to any other steps in the boot sequence
+##      - ${STARTUP_USER_SCRIPTS} are executed after ogg/nginx startup
+##
+## When ${ABORT_ON_USER_SCRIPT_ERRORS} is true:
+##    Scripts must return a status of 0 on exit or the boot proccess will abort
+##
+## When ${ABORT_ON_USER_SCRIPT_ERRORS} is false:
+##    Scripts that return a non-zero status will issue a warning but will continue
+##    the boot process
+##
+function run_user_scripts {
+
+    local SCRIPTS_ROOT="${1}";
+    
+    [ -z "$SCRIPTS_ROOT" ] && {
+        printf "%s: No SCRIPTS_ROOT passed on, no scripts will be run\n" "${0}";
+        return 1;
+    } || {
+        if [ -d "$SCRIPTS_ROOT" ] && [ -n "$(ls -A "$SCRIPTS_ROOT")" ]; then
+            printf "Executing user defined scripts in: %s\n" "${SCRIPTS_ROOT}"
+        
+            for f in "${SCRIPTS_ROOT}"/*; do
+                [ -d "${f}" ] && {
+                    run_user_scripts "${f}"
+                } || {
+                    case "$f" in
+                        *.sh)
```

Comment:
I would vote for executing anything with the executable bit set. That way, Python scripts (or whatever) could also be run. But, the Database only supports `*.sh` files, so maybe we should stick to just those.

Reply:
I really like the idea of checking the execute bit.  The issue I worry about there is the oddities we see in git check ins where those aren't stored correctly.

I will defer to your call here.

Reply:
Let's stick to the DBMS model and only process `*.sh` files via `source` (or `.`)

Reply:
replaced with the source for processing *.sh files